Common/Securities/SecurityPortfolioManager.cs

```diff
@@ -660,7 +681,92 @@ public void ApplySplit(Split split)
         }
+        /// <summary>
+        /// Applies a split to the portfolio equity options positions
+        /// </summary>
+        /// <param name="split">The split to be applied</param>
+        private void ApplySplitToOptions(Split split)
+        {
+            // only apply to the option positions that have correct underlying symbol
+            var optionSecurities = Securities
+                                   .Where(x => x.Value.Type == SecurityType.Option && split.Symbol == x.Key.Underlying && x.Value.Holdings.Invested)
+                                   .ToList();
+
+            foreach (var securityKV in optionSecurities)
+            {
+                var symbol = securityKV.Key;
+                var security = securityKV.Value;
+
+                // only apply splits in raw data mode, 
+                var mode = security.DataNormalizationMode;
+                if (mode != DataNormalizationMode.Raw)
+                {
+                    continue;
+                }
+
+                var splitFactor = split.SplitFactor;
+                var newSymbol = GetSplitAdjustedSymbol(symbol, splitFactor);
 
+                if (newSymbol != null)
+                {
+                    Securities.Remove(symbol);
+                    var optionHoldings = new Option.OptionHolding(Securities[newSymbol], (Option.OptionHolding)security.Holdings);
+                    optionHoldings.SplitUnderlying(splitFactor);
+                    Securities[newSymbol].Holdings = optionHoldings;
+                }
+                else
+                {
+                    var optionHoldings = (Option.OptionHolding)security.Holdings;
+                    optionHoldings.SplitUnderlying(splitFactor);
+                }
+            }
+        }
+
+        private Symbol GetSplitAdjustedSymbol(Symbol symbol, decimal splitFactor)
+        {
+            var inverseFactor = 1.0m / splitFactor;
+
+            decimal newStrike = 0.0m;
+            string newRootSymbol = null;
+
+            Func<Symbol, bool> symbolIsFound = x => 
+            {
+                var rootSymbol = newRootSymbol ?? symbol.Underlying.Value;
+                var strike = newStrike != 0.0m ? newStrike : symbol.ID.StrikePrice;
+
+                return x.HasUnderlying == true &&
+                        x.Underlying.Value == rootSymbol &&
+                        x.ID.Date == symbol.ID.Date &&
+                        x.ID.OptionRight == symbol.ID.OptionRight &&
+                        x.ID.Market == symbol.ID.Market &&
+                        x.ID.OptionStyle == symbol.ID.OptionStyle &&
+                        x.ID.StrikePrice == strike;
+            };
+
+            // detect forward (even and odd) and reverse splits
+            if (splitFactor > 1.0m)
+            {
+                // reverse split
+                newRootSymbol = symbol.Underlying.Value + "1";
+            }
+            if ((int)Math.Round(inverseFactor, 5) == (int)inverseFactor)
```

Comment:
That was to distinguish even splits (2 for 1) from odd ones (3 for 2). Updated the code.